File: coko/com.elster.jupiter.metering.rest/src/main/java/com/elster/jupiter/metering/rest/impl/LocationInfo.java
/*
 * Copyright (c) 2017 by Honeywell International Inc. All Rights Reserved
 */

package com.elster.jupiter.metering.rest.impl;

import java.util.ArrayList;
import java.util.List;

public class LocationInfo {
    public long id;
    public String name;
    public List<LocationMemberInfo> members = new ArrayList<>();

    public LocationInfo() {
    }
}

File: copl/com.elster.jupiter.unifyingjs/src/main/web/js/uni/src/model/LocationInfo.js
/*
 * Copyright (c) 2017 by Honeywell International Inc. All Rights Reserved
 */

Ext.define('Uni.model.LocationInfo', {
    extend: 'Ext.data.Model',
    requires: [
        'Uni.property.model.Property'
    ],

    fields: [
        'locationId', 'locationValue', 'unformattedLocationValue', 'usagePointLocationId', 'isInherited'
    ],
    associations: [
        {
            name: 'properties',
            type: 'hasMany',
            model: 'Uni.property.model.Property',
            associationKey: 'properties',
            foreignKey: 'properties',
            getTypeDiscriminator: function (node) {
                return 'Uni.property.model.Property';
            }
        }
    ],
    proxy: {
        type: 'rest',
        reader: {
            type: 'json'
        }
    }


});